drivers/bluetooth/hci/userchan.c

```diff
@@ -93,6 +95,60 @@ static struct net_buf *get_rx(const uint8_t *buf)
 	return NULL;
 }
 
+/**
+ * @brief Decode the length of an HCI H4 packet
+ * @details Decodes packet length according to Bluetooth spec v5.2 Vol 4 Part E
+ * @param packet_buff	Pointer to a HCI packet buffer
+ */
+static uint16_t packet_len(const uint8_t *packet_buff)
+{
+	uint16_t payload_len = 0;
+	uint8_t header_len = 0;
+	const uint8_t packet_type = packet_buff[0];
+
+	switch (packet_type) {
+	case H4_CMD:
+		const struct bt_hci_cmd_hdr *hdr = &packet_buff[sizeof(packet_type)];
```

Comment:
```suggestion
	const uint8_t packet_type = packet_buff[0];
	const uint8_t *packet_hdr = &packet_buff[sizeof(packet_type)];

	switch (packet_type) {
	case H4_CMD:
		const struct bt_hci_cmd_hdr *hdr = packet_hdr;
```

Since they all share the same offset, consider making doing it a single time.